docs/week-3.md

```diff
@@ -0,0 +1,523 @@
+```
+
+At the top of our file we import the React library. We then start by building a functional component 
+(which we will explain in just a moment) to display our lucky number to the user. Inside our component we return 
+the JSX (Javascript XML) we want to be displayed inside the return statement. We then export the component for use 
+in other places.
+
+
+## Components
+
+Components are the building blocks of a React application. Understanding what one is and how they are important is 
+fundamental to one's comprehension of React.
+
+### What is a Component?
+A React component is like a piece of the "React App" puzzle. For a puzzle to be complete we need at least one piece. 
+Similarly, each React application needs at least one component. The default component is usually one by the name of 
+App.js or in our case App.tsx (we are using Typescript instead of Javascript). Let us take a look at the LuckyNum.tsx
+file created above.
+```
+import React from 'react'; 
+
+const LuckyNum = () => {
+
+    let myLuckyNumber = 7;
+
+    return(
+        <div>
+            <h1>My Lucky Number is {myLuckyNumber}</h1>
+        </div>
+    );
+}
+
+export default LuckyNum;
+```
+Here we see the creation of a component. The component's name is LuckyNum and displays our lucky number in a `<h1>`
+tag. We know this is a component because it is returning JSX to be displayed to a user. Notice how our `<h1>` is
+wrapped in a `<div>`. This is because React requires us to return a contained component. This means that there most be
+a type of container that has what we want to display housed inside. In most cases using a `<div>` is the best way to 
+go. However, you can use any piece of JSX as a wrapper/container.
+
+So, we have covered that a component is just a piece of a React App. A cool thing we can do in react is that we can
+use components that we have made in other components that we make.
+
+### Using Components
+Remember our LuckyNum component we made above? We are going to use this component in our Profile component below in
+order to display your favorite number along with your name and favorite color.
+
+```
+import React from 'react';
+import LuckyNum from './LuckyNum';
+
+/* 
+   Notice how there is no return statement. If we are only 
+   returning JSX (no javascript inside of our component) 
+   we can use () to wrap our component
+*/
+
+const Profile = () => (
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+    </div>
+);
+
+export default Profile;
+```
+
+As you can see we import the component `LuckNum` from the typescript file we wrote. Once the component is imported
+we are able to use it in our new component. The syntax to do this is by creating a tag to indicate where we
+want the imported component to be placed.
+
+## Props
+Looks like things are going well. We have successfully displayed John's profile picture, name, favorite color and
```

Comment:
Oxford comma? (I don't know if we're really that set on enforcing this though so you can probably leave it. @sadafkh11 thoughts?)

```suggestion
## Props

Looks like things are going well. We have successfully displayed John's profile picture, name, favorite color, and
```